apps/emqx/src/emqx_authentication.erl

```diff
@@ -649,7 +637,7 @@ handle_create_authenticator(Chain, Config, Providers) ->
     end.
 
 do_authenticate(_ChainName, [], _) ->
-    {stop, {error, not_authorized}};
+    {ok, {error, not_authorized}};
```

Comment:
These two changed lines in `do_authenticate/3` is the major part of the fix. We do not stop the hook chain fold, but return the new acc.
In case there are no other hooks, the behaviour is the same.
In case there are other hooks with lower priority, the can override the auth result (acc).

Reply:
should return ignore from here ?

Reply:
Seems that no, because it will change the behaviour. Currently if we have several providers all of which return `ignore`, authentication fails.
Returning `ignore` here will make authentication successful.